otomoto/spiders/otomoto.py

```diff
@@ -16,14 +16,24 @@ def remove_spaces(x):
 
 
 def convert_to_integer(x):
-    return int(x)
-
+    y = x.split(',', 1)[0]
+    return int(y)
+
+def convert_vin(x):
+    if "?vin=" in x:
+        y = (x.split('?vin=', 1)[1]).split('#utm', 1)[0].split('&', 1)[0]
+        if len(y) != 17:
+            y = 'null'
+    else:
+        y = 'null'
+    return y
```

Comment:
Can you use `urlparse`.  Function will be much cleaner.  I think you can also take an advantage of the fact that vin number is 17 digit always.

Probably `urlparse` will be much cleaner solution.